crates/cli/src/updater.rs

```diff
@@ -300,18 +301,16 @@ impl Updater {
             let app_string = app.get_base_name();
             let current_binary = self.binaries.get(&app_string).cloned();
             tasks.push(tokio::spawn(async move {
-                let (_, info_url) = match get_release_url(app, None, None).await {
-                    Ok(urls) => urls,
-                    Err(_) => return,
-                };
-                let manifest = match fetch_manifest(&info_url, app).await {
-                    Ok(manifest) => manifest,
-                    Err(_) => return,
-                };
-                if let Some(current_manifest) = current_binary {
-                    if manifest.release != current_manifest.release && manifest.release.is_some() {
+                {
+                    let (result, version) = match app {
+                        SupportedApp::Marzano | SupportedApp::Gouda => {
+                            check_release_axo(app, &current_binary).await.unwrap()
+                        }
+                        _ => check_release_internal(app, &current_binary).await.unwrap(),
+                    };
+                    if result {
                         let mut found = found_updates.lock().await;
-                        found.push((app, manifest.version.unwrap()));
+                        found.push((app, version.unwrap()));
```

Comment:
Since this method returns a Result<> we should avoid unwrap - can you refactor to remove?

Reply:
Sure thing! I've pushed a commit which replicates the original logic, with a `return` if an `Err` is encountered when checking for a release. This is within an `async` block, which wasn't itself returning `Result<>`, so the question mark operator was trickier.

The `version.unwrap()`, which is on an `Option<>` is left alone from how it was before this PR, but let me know if you'd like me to adjust that as well.

Reply:
Now that I think about it, eliminating the `version.unwrap()` doesn't add too much more complexity, so pushed a change that does that as well.

Reply:
Great, thanks for improving it.